### PR TITLE
refactor:simplified instantiation of OID4VCIssuerEndpoint

### LINKED ISSUE #34762: [OID4VCI] Refactor the OID4VCLoginProtocolFactory to follow the pattern in OIDCLoginProtocolFactory & OIDCLoginProtocolService
### Description

**Refactor the `OID4VCLoginProtocolFactory` to align with the structure used in `OIDCLoginProtocolFactory` and `OIDCLoginProtocolService`. This will enable the capability to implement the `credential` endpoint as part of the `openid-connect` protocol.**

As part of implementing a scope-based approach, we plan to remove the `oid4vc` protocol and treat the `credential` endpoint as a sub-protocol of OAuth2.

- **Current**: `https://keycloak-demo.solutions.adorsys.com/realms/master/protocol/**oid4vc**/credential`
- **Future**: `https://keycloak-demo.solutions.adorsys.com/realms/master/protocol/**openid-connect**/credential`

This ticket will not involve refactoring the `oidc` protocol endpoint itself but will streamline the code for easier refactoring in the future.

**References**:
- #32951
- #32953

### PR DESCRIPTION
**Pull request by francis-pouatcha**
Simplified instantiation if OID4VCIssuerEndpoint, such that it can be integrated ino the OIDC protocol.

No need for additional tests, as simple transfer of method from OID4VCLoginProtocolFactory to OID4VCIssuerEndpoint.

Closes #34762

### CODE REVIEW

**Comment by tnorimat** [comment]
@francis-pouatcha Hello, it seems that the PR lacks DCO. Could you re-push the PR with a commit "-s" option?

**Review by tnorimat — COMMENTED** [review]
Hello, I added one review comment. Could you check it?
Also please add your signature to the commit by following:
https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md#developers-certificate-of-origin

File: services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java [hunk 1]
```diff
@@ -164,6 +172,39 @@ public OID4VCIssuerEndpoint(KeycloakSession session,
         this.isIgnoreScopeCheck = isIgnoreScopeCheck;
     }
 
+     public OID4VCIssuerEndpoint(KeycloakSession keycloakSession, EventBuilder event){
```
[the remark this answers is not in the document]
Comment (francis-pouatcha) [reply services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerEndpoint.java:175]: Thank you, I will remove it and reinstate it when the endpoint utilizes it.

**Comment by mposolda** [comment]
@tnorimat Can you please review this PR? I can merge it once you approve the PR.

**Review by tnorimat — APPROVED** [review]
LGTM

**Comment by tnorimat** [comment]
@mposolda Hello, I approved the PR.

**Review by mposolda — APPROVED** [review]
@francis-pouatcha @tnorimat Thanks!

**Pull request merged**